core/src/main/scala/sttp/tapir/Codec.scala

```diff
@@ -87,6 +87,22 @@ trait Codec[L, H, +CF <: CodecFormat] { outer =>
   def mapDecode[HH](f: H => DecodeResult[HH])(g: HH => H): Codec[L, HH, CF] = map(Mapping.fromDecode(f)(g))
   def map[HH](f: H => HH)(g: HH => H): Codec[L, HH, CF] = mapDecode(f.andThen(Value(_)))(g)
 
+  /** Maps this codec to the given higher-level type `HH`.
+    *
+    * @param f
+    *   decoding function
+    * @param g
+    *   encoding function
+    * @tparam HH
+    *   target type
+    * @see
+    *   [[map]]
+    * @see
+    *   [[mapDecode]]
+    */
+  def emap[HH](f: H => Either[String, HH])(g: HH => H): Codec[L, HH, CF] =
```

Comment:
thanks for the proposition :) I'm wondering about the naming - since we already have `mapDecode`, `mapValidate`, wouldn't it be more consistent to use e.g. `mapEither`?

Reply:
Makes sense! I renamed it